tests/test_utils.py

```diff
@@ -0,0 +1,124 @@
+import pytest
+from textwrap import dedent
+from pipeline_migration.utils import YAMLStyle, dump_yaml, BlockSequenceIndentation
+
+YAML_EXAMPLE_0_INDENT = """\
+apiVersion: tekton.dev/v1
+spec:
+  params:
+  - name: git-url
+    type: string
+  - name: revision
+    type: string
+  tasks:
+  - name: clone-repository
+  - name: build-container
+"""
+
+YAML_EXAMPLE_2_INDENTS = """\
+apiVersion: tekton.dev/v1
+spec:
+  params:
+    - name: git-url
+      type: string
+    - name: revision
+      type: string
+  tasks:
+    - name: clone-repository
+    - name: build-container
+"""
```

Comment:
Could we have a simple "round-trip" test that loading and dumping these yamls results in the same output?

We have unit tests that separately test block indentation detection and dumping a yaml with a specified block indentation, but I don't think we have a round-trip test that would clearly demonstrate that the format is preserved

Reply:
https://github.com/konflux-ci/pipeline-migration-tool/pull/20/files#diff-4e8715c7a425ee52e74b7df4d34efd32e8c92f3e60bd51bc2e1ad5943b82032eR231

How about this test? The fixture `pipeline_yaml_with_various_indent_styles` is parameterized with YAMLs with different indentation levels.

Reply:
Those test that YAMLStyle detection before/after returns the same results, but as far as I can tell, they don't really test that the yaml content really looks the same